### PR TITLE
feat: add math using alternative way to include css

### PR DESCRIPTION
**Pull request by cabljac**
This PR adds support for maths in docs using remark-katex. To enable, you simply add the flag:

`experimentalMath: true`

in the docs.json. The extra css for Katex is injected in **only if**  this flag is set. 

Screenshot 2022-03-11 at 10 39 48

### CODE REVIEW

File: website/app/routes/$owner.$repo.custom-styles[.]css.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { LoaderFunction } from 'remix';
+
+async function getConfig({ owner, repo }: Record<string, string>) {
+  const endpoint = `https://raw.githubusercontent.com/${owner}/${repo}/main/docs.json`;
+  const res = await (await fetch(endpoint)).text();
+
+  return JSON.parse(res);
+}
+
+export let loader: LoaderFunction = async ({ request, params }) => {
```
Comment (cabljac) [inline website/app/routes/$owner.$repo.custom-styles[.]css.ts:10]:
```suggestion
export const loader: LoaderFunction = async ({ request, params }) => {
```

File: website/app/routes/$owner.$repo.custom-styles[.]css.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import type { LoaderFunction } from 'remix';
+
+async function getConfig({ owner, repo }: Record<string, string>) {
```
Comment (cabljac) [inline website/app/routes/$owner.$repo.custom-styles[.]css.ts:3]: This probably needs moving/refactoring, and integrating into our type validation stuff

File: api/src/controllers/github.ts [hunk 3]
```diff
@@ -5,8 +5,6 @@ import { getGitHubContents, getPullRequestMetadata } from '../utils/github.js';
 import { HeadingNode } from '../utils/plugins/rehype-headings.js';
 import { getPlugins } from '../utils/getPlugins.js';
 import { getFilePath, getRepositorySymLinks } from '../utils/symlinks.js';
```
Comment (cabljac) [inline api/src/controllers/github.ts:7]:
```suggestion
import { getRepositorySymLinks } from '../utils/symlinks.js';
```

File: website/app/routes/$owner.$repo.custom-styles[.]css.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  const endpoint = `https://raw.githubusercontent.com/${owner}/${repo}/main/docs.json`;
```
Comment (Salakar) [inline website/app/routes/$owner.$repo.custom-styles[.]css.ts:4]: Should Ref not be taken into account here?

Comment (cabljac) [reply website/app/routes/$owner.$repo.custom-styles[.]css.ts:4]: yeah for sure.

**Comment by cabljac** [comment]
Also - gives errors in the console about not being able to find font files locally, the maths still renders though

File: website/app/routes/$owner.$repo.custom-styles[.]css.ts (hunk 4 above)
Comment (cabljac) [reply website/app/routes/$owner.$repo.custom-styles[.]css.ts:4]: Also I should integrate it with the ProjectConfig type, and provide a default, and probably move it to the config utils file

**Comment by Salakar** [comment]
Can we fix formatting issues

**Comment by cabljac** [comment]
Note this also needs to be implemented for preview mode, shouldn't be difficult to do

**Pull request merged**